pkg/sql/zone_config.go

```diff
@@ -279,6 +279,59 @@ func GetZoneConfigInTxn(
 	return zoneID, zone, subzone, nil
 }
 
+// GetHydratedZoneConfigForTable returns a fully hydrated zone config for a
+// given descriptor ID.
```

Comment:
s/descriptor ID/table ID. We need to write tests for this if not too inconvenient to graft from elsewhere in this package. Ideally something to also capture partitions. Is there a CCL variant of zone config tests we could steal from?